test/logstream/v2/stream.go

```diff
@@ -34,21 +34,25 @@ import (
 	"knative.dev/pkg/ptr"
 )
 
-func FromNamespaces(ctx context.Context, c kubernetes.Interface, namespaces []string) Source {
+func FromNamespaces(ctx context.Context, c kubernetes.Interface, namespaces []string, filterLines bool, podPrefixes ...string) Source {
 	return &namespaceSource{
-		ctx:        ctx,
-		kc:         c,
-		namespaces: namespaces,
-		keys:       make(map[string]Callback, 1),
+		ctx:         ctx,
+		kc:          c,
+		namespaces:  namespaces,
+		keys:        make(map[string]Callback, 1),
+		filterLines: filterLines,
+		podPrefixes: podPrefixes,
 	}
 }
 
-func FromNamespace(ctx context.Context, c kubernetes.Interface, namespace string) Source {
+func FromNamespace(ctx context.Context, c kubernetes.Interface, namespace string, filterLines bool, podPrefixes ...string) Source {
```

Comment:
This is a breaking API change. You could extend this method without breaking by using variadic functional options

see option 3 - https://golang.cafe/blog/golang-functional-options-pattern.html

Reply:
Thanks. I have tried what you're suggesting and pushed a new commit. Please let me know what you think. I think this new way of constructing the logstream could replace the "FromNamespace" and "FromNamespaces" functions so we could mark them as deprecated (which I haven't done yet).

Reply:
I have also vendored the changes into https://github.com/knative/serving/pull/13293 so the exaple usage can be seen there.